src/server-extension/resolvers/series.ts

```diff
@@ -61,12 +76,20 @@ export class SeriesResolver {
     return result
   }
 
-  @FieldResolver(() => [HistoryEntity])
-  async buyHistory(@Root() series: SeriesEntity) {
-
+  @Query(() => [HistoryEntity])
+  async seriesInsightBuyHistory(
+    @Arg('ids', () => [String!], { nullable: false }) ids: CollectionIDs,
+    @Arg('dateRange', { nullable: false, defaultValue: '7 DAY' }) dateRange: DateRange,
+  ) {
+    const idList = JSON.stringify(ids).replace(/\"/g, '\'').replace(/[\[|\]]/g, '')
```

Comment:
Afaik if you use that as a dollar param you dont need to do this

Reply:
answered at [here](https://github.com/kodadot/rubick/pull/72#discussion_r873325836)

Reply:
please move the function to utils